src/components/AttachmentModal.tsx

```diff
@@ -1,186 +1,218 @@
+    onModalShow?: () => void;
 
-    ...withLocalizePropTypes,
+    /** Optional callback to fire when we want to do something after modal hide. */
+    onModalHide?: () => void;
 
-    ...windowDimensionsPropTypes,
+    /** Optional callback to fire when we want to do something after attachment carousel changes. */
+    onCarouselAttachmentChange?: (attachment: Attachment) => void;
 
     /** Denotes whether it is a workspace avatar or not */
-    isWorkspaceAvatar: PropTypes.bool,
+    isWorkspaceAvatar?: boolean;
 
     /** Whether it is a receipt attachment or not */
-    isReceiptAttachment: PropTypes.bool,
-};
+    isReceiptAttachment?: boolean;
 
-const defaultProps = {
-    source: '',
-    onConfirm: null,
-    defaultOpen: false,
-    originalFileName: '',
-    children: null,
-    isAuthTokenRequired: false,
-    allowDownload: false,
-    headerTitle: null,
-    report: {},
-    transaction: {},
-    onModalShow: () => {},
-    onModalHide: () => {},
-    onCarouselAttachmentChange: () => {},
-    isWorkspaceAvatar: false,
-    isReceiptAttachment: false,
+    /** A function as a child to pass modal launching methods to */
+    children?: React.FC<ChildrenProps>;
+
+    fallbackSource?: AvatarSource | number;
 };
 
-function AttachmentModal(props) {
-    const theme = useTheme();
+function AttachmentModal({
+    source = '',
+    onConfirm = null,
+    defaultOpen = false,
+    originalFileName = '',
+    isAuthTokenRequired = false,
+    allowDownload = false,
+    report,
+    onModalShow = () => {},
+    onModalHide = () => {},
+    onCarouselAttachmentChange = () => {},
+    isReceiptAttachment = false,
+    isWorkspaceAvatar = false,
+    transaction,
+    parentReport,
+    parentReportActions,
+    headerTitle,
+    policy,
+    children,
+    fallbackSource,
+}: AttachmentModalProps) {
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
-    const [isModalOpen, setIsModalOpen] = useState(props.defaultOpen);
+    const [isModalOpen, setIsModalOpen] = useState(defaultOpen);
     const [shouldLoadAttachment, setShouldLoadAttachment] = useState(false);
     const [isAttachmentInvalid, setIsAttachmentInvalid] = useState(false);
+
     const [isDeleteReceiptConfirmModalVisible, setIsDeleteReceiptConfirmModalVisible] = useState(false);
-    const [isAuthTokenRequired, setIsAuthTokenRequired] = useState(props.isAuthTokenRequired);
+    const [isAuthTokenRequiredState, setIsAuthTokenRequiredState] = useState(isAuthTokenRequired);
     const [attachmentInvalidReasonTitle, setAttachmentInvalidReasonTitle] = useState('');
-    const [attachmentInvalidReason, setAttachmentInvalidReason] = useState(null);
-    const [source, setSource] = useState(props.source);
-    const [modalType, setModalType] = useState(CONST.MODAL.MODAL_TYPE.CENTERED_UNSWIPEABLE);
+    const [attachmentInvalidReason, setAttachmentInvalidReason] = useState<string | null>(null);
```

Comment:
It should help to get rid of assertion
```suggestion
    const [attachmentInvalidReason, setAttachmentInvalidReason] = useState<TranslationPaths | null>(null);
```